packages/NodeCollector/StaticAnalyzer.php

```diff
@@ -45,7 +45,7 @@ private function hasStaticAnnotation(string $methodName, ClassReflection $classR
 
         return (bool) Strings::match(
             $resolvedPhpDocBlock->getPhpDocString(),
-            '#@method\s*static\s*(.*?)\b' . $methodName . '\b#'
+            '#@method\s*static\s*((([\w\|\\\\]+)|\$this)*+(\[\])*)*\s+\b' . $methodName . '\b#'
```

Comment:
Thanks!

Could you add a link to https://regex101.com/ with example for this regex? 
It would be easier to maintain in the future

Reply:
Updated